### ISSUE TITLE
Implement support for exporting Chapel arrays, part 1

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#9600]
This is intended to cover supporting module code changes.  I need to make a subclass of arrays/domains/etc. that recognizes it does not own its contents and so doesn't not delete its _ddata when it goes out of scope.  This will likely benefit from similar changes to ArrayViewSlice*

**Comment by lydia-duncan** [comment]
To test this, I think I want to make a little program that calls an extern function to get an allocated C array back, and then wraps that returned array in this wrapper.  Then, I should be able to call the extern free routine on it and not get a double delete when the program ends.

**Comment by lydia-duncan** [comment]
I suspect I want to implement the these() iterators, dsiSerialRead/Write, displayRepresentation, access and check bounds functions right away.

I may need dsiGetBaseDom right away as well but I am going to wait on it for now.  I will definitely need the locality, privatization, bulk transfer stuff eventually, but am not going to worry about them right away.

**Comment by lydia-duncan** [comment]
Discussed a bug along the way with Brad, end result being that I need to define a domain class as well

**Comment by lydia-duncan** [comment]
Figuring out read/write stuff at the moment.

**Comment by lydia-duncan** [comment]
Later TODOs:

- [ ] locality?
- [ ] privatization
